python/pyproject.toml

```diff
@@ -254,7 +254,11 @@ shell = "printf '%s\\n' $TARGET | grep -E '\\.(py|ipynb)$' | tr '\\n' ' '"
 
 [tool.poe.tasks.type-check]
 help = "Static Type Checker with optional flags"
-cmd = "mypy . ${POE_ARGS}"
+args = [
+    { name = "TARGET", positional = true, multiple = true, default = "." }
+]
+uses = { PY_FILES = "_extract_python_files $TARGET" }
+cmd = "mypy ${PY_FILES:-.} ${POE_ARGS}"
```

Comment:
I tried this locally (with my uncommitted changes for my ongoing work) and it still just ran `mypy .` not `mypy <list of files>` also from my reading of this `_extract_python_files` just gets a list of the py files in the dir, not ones with changes.

Also after looking at the logs, I'm not sure `_extract_python_files` is ever returning a list of files, because lint and format also run with `PY_FILES` empty

Reply:
pre-commit passes changed files

Reply:
I'll test it with the pre-commit when doing my upcoming commit to 583

Reply:
does `_extract_python_files` not check py files where only the imports are edited?

because in a commit where i edited 6 files it only caught four. One of the two it missed I only moved it's dir, but the other I moved and edited the imports

Reply:
also this tries running mypy on edited ipynb files, which will cause errors

Reply:
Regarding file matching, you must delve into `.githooks/hook.sh`. That script is responsible for detecting changes and calling the pre-commit script.